File: app/src/main/java/com/example/nasmovie/data/model/Favorite.java
package com.example.nasmovie.data.model;

import androidx.annotation.NonNull;
import androidx.room.Entity;
import androidx.room.Ignore;
import androidx.room.PrimaryKey;

/**
 * 收藏实体
 */
@Entity(tableName = "favorite")
public class Favorite {

    @NonNull
    @PrimaryKey
    private String movieId = "";      // 电影ID
    private long addTime;           // 收藏时间

    public Favorite() {
        this.addTime = System.currentTimeMillis();
    }

    @Ignore
    public Favorite(String movieId) {
        this.movieId = movieId;
        this.addTime = System.currentTimeMillis();
    }

    // Getters and Setters
    public String getMovieId() {
        return movieId;
    }

    public void setMovieId(String movieId) {
        this.movieId = movieId;
    }

    public long getAddTime() {
        return addTime;
    }

    public void setAddTime(long addTime) {
        this.addTime = addTime;
    }
}

File: app/src/main/java/com/example/nasmovie/data/model/Favorite.kt
package com.example.nasmovie.data.model

import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.Ignore
import androidx.room.Index
import androidx.room.PrimaryKey

/**
 * 收藏实体
 */
@Entity(
    tableName = "favorite",
    indices = [Index("addTime"), Index("movieId")],
    foreignKeys = [
        ForeignKey(
            entity = Movie::class,
            parentColumns = ["id"],
            childColumns = ["movieId"],
            onDelete = ForeignKey.CASCADE,
            onUpdate = ForeignKey.CASCADE
        )
    ]
)
data class Favorite(
    @PrimaryKey
    var movieId: String = "",
    var addTime: Long = System.currentTimeMillis()
) {
    @Ignore
    constructor(movieId: String) : this(movieId, System.currentTimeMillis())
}